### PR TITLE
FAN-1855 Remove memcache usage for User Attributes

### PR DESCRIPTION
**Pull request by mklucsarits**

### CODE REVIEW

File: lib/Wikia/src/Service/User/Attributes/UserAttributes.php [hunk 1]
```diff
@@ -78,30 +70,18 @@ private function loadAttributes( $userId ) {
...
-				$attributes[$attribute->getName()] = $attribute->getValue();
-			};
-			$this->logAttributeServiceRequest( $userId );
-			$this->setInMemcache( $userId, $attributes );
-		}
+		$attributes = [];
+		/** @var Attribute $attribute */
+		foreach ( $this->attributeService->get( $userId ) as $attribute ) {
+			$attributes[$attribute->getName()] = $attribute->getValue();
+		};
```
Comment (WikiaTech) [inline lib/Wikia/src/Service/User/Attributes/UserAttributes.php]: MINOR Remove this empty statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1116)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [UserAttributes.php#L27](https://github.com/Wikia/app/blob/a006dfc9866732c60dc36a5b4770ad9c3ed22fa7/lib/Wikia/src/Service/User/Attributes/UserAttributes.php#L27): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [UserAttributes.php#L103](https://github.com/Wikia/app/blob/a006dfc9866732c60dc36a5b4770ad9c3ed22fa7/lib/Wikia/src/Service/User/Attributes/UserAttributes.php#L103): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [UserAttributes.php#L27](https://github.com/Wikia/app/blob/ee7293d4085c9aff2a1a632f495a7386340ec7be/lib/Wikia/src/Service/User/Attributes/UserAttributes.php#L27): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [UserAttributes.php#L103](https://github.com/Wikia/app/blob/ee7293d4085c9aff2a1a632f495a7386340ec7be/lib/Wikia/src/Service/User/Attributes/UserAttributes.php#L103): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Comment by mszabo-wikia** [comment]
@mklucsarits With the removal of `BagOStuffCacheProvider`, do we still rely on `doctrine/cache` anywhere? Could we remove this composer dependency?

**Comment by jsutterfield** [comment]
@TK-999 looks like the preference service client is still using that code... https://github.com/Wikia/app/blob/dev/lib/Wikia/src/Service/User/Preferences/PreferenceModule.php#L30

**Review by jsutterfield — CHANGES_REQUESTED** [review]
LGTM, although looks like we have tests failing here: https://github.com/Wikia/app/blob/dev/lib/Wikia/tests/Service/User/UserAttributesTest.php

Those `UserAttributes` objects are being created with a mocked cache so the signatures are out of date. We should be able to just remove those and I think it'll work.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [UserAttributes.php#L27](https://github.com/Wikia/app/blob/b7a283de32d36ad82e8c84d30ed877f15b76b315/lib/Wikia/src/Service/User/Attributes/UserAttributes.php#L27): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [UserAttributes.php#L103](https://github.com/Wikia/app/blob/b7a283de32d36ad82e8c84d30ed877f15b76b315/lib/Wikia/src/Service/User/Attributes/UserAttributes.php#L103): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [UserAttributes.php#L101](https://github.com/Wikia/app/blob/515db3317cc1e94af08befaa7779a1685c004a49/lib/Wikia/src/Service/User/Attributes/UserAttributes.php#L101): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Review by jsutterfield — APPROVED** [review]

**Comment by jsutterfield** [comment]
LGTM!

**Pull request merged**